### PR TITLE
Fix e2e network performance test

### LINKED ISSUE #90450: E2E: networking_perf.go test erroring out
**What happened**:

E2E:networking_perf.go test failing
My cluster is based on 1.16.3 release with multus CNI, when I ran networking_perf suite from e2e.test, it is failing out with below error.
Apr 17 07:46:25.532: INFO: At 2020-04-17 07:36:40 +0000 UTC - event for iperf-e2e-pod-0: {kubelet worker-pool1-09u5e8jl-esiperu-demo-cluster} FailedCreatePodSandBox: Failed create pod sandbox: rpc
error: code = Unknown desc = failed to set up sandbox container "fb85ec94a3d2f884902dc4ccc5cf05cf9c50bb4f53d4f14903fcad3302fd18d8" network for pod "iperf-e2e-pod-0": networkPlugin cni failed to set
 up pod "iperf-e2e-pod-0_network-perf-3964" network: Multus: [network-perf-3964/iperf-e2e-pod-0]: error adding container to network "k8s-pod-network": delegateAdd: cannot set "" interface name to "
eth0": validateIfName: no net namespace /proc/29818/ns/net found: failed to Statfs "/proc/29818/ns/net": no such file or directory 

**What you expected to happen**:
The networking_perf test should work.

**How to reproduce it (as minimally and precisely as possible)**:
 ./hack/ginkgo-e2e.sh --ginkgo.focus=Feature:Networking-Performance --ginkgo.skip=IPv4

**Anything else we need to know?**:

This issue is there in current master branch also

Both IPv4 and IPv6 Fails.
The issue is container configuration in the test, the ENTRYPOINT for the agnhost image is agnhost ,
which will not take the argument as ""/bin/sh",
								"-c",
								"/usr/local/bin/iperf " + familyStr + "-c service-for-" + appName + " -p 8002 --reportstyle C && sleep 5"," hence container is coming out early, thus CNI is complaining.

container configuration should take command as "/bin/sh" and argument as "-c",
								"/usr/local/bin/iperf " + familyStr + "-c service-for-" + appName + " -p 8002 --reportstyle C && sleep 5"

.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.3
- Cloud provider or hardware configuration: 
- OS (e.g: `cat /etc/os-release`): SUSE Linux Enterprise Server 15 SP1
- Kernel (e.g. `uname -a`):4.12.14-197.29-default
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/sig network
/sig testing

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation

/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:

The agnhost image has the `agnhost` binary as entrypoint.
The networking performance test need to run iperf overriding the entrypoint

**Which issue(s) this PR fixes**:

Fixes #90450

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by aojea** [comment]
I've tested it locally, don't know if this is running in some CI

```
bazel-bin/test/e2e/e2e.test -ginkgo.focus="Feature:Networking-Performance" -num-nodes 2
....
Apr 25 16:42:24.881: INFO: Selector matched 1 pods for map[app:iperf-e2e-cli-pod]
Apr 25 16:42:24.881: INFO: ForEach: Found 1 pods from the filter.  Now looping through them.
Apr 25 16:42:24.881: INFO: Running '/usr/local/bin/kubectl --server=https://127.0.0.1:35859 --kubeconfig=/home/aojeagarcia/.kube/config logs iperf-e2e-cli-pod-0 iperf-client --namespace=network-perf-7184'
Apr 25 16:42:24.999: INFO: stderr: ""
Apr 25 16:42:24.999: INFO: stdout: "20200425144219,10.244.1.7,55390,10.103.237.250,8002,3,0.0-10.0,41231187968,32984805241\n"
Apr 25 16:42:24.999: INFO: 20200425144219,10.244.1.7,55390,10.103.237.250,8002,3,0.0-10.0,41231187968,32984805241
....
Ran 2 of 5009 Specs in 50.958 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 5007 Skipped
```

File: test/e2e/network/networking_perf.go [hunk 1]
```diff
@@ -70,10 +70,10 @@ func networkingIPerfTest(isIPv6 bool) {
 			func(n v1.Node) v1.PodSpec {
 				return v1.PodSpec{
 					Containers: []v1.Container{{
-						Name:  "iperf-server",
-						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+						Name:    "iperf-server",
+						Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+						Command: []string{"/bin/ash"},
```
Comment (SataQiu) [inline test/e2e/network/networking_perf.go:75]: @aojea  This generally SGTM.
But why not use `/bin/sh` but `/bin/ash`?

```sh
# docker run -it --entrypoint ls  us.gcr.io/k8s-artifacts-prod/e2e-test-images/agnhost:2.13 /bin/
ash            dnsdomainname  ipcalc         mv             sh
base64         dumpkmap       kbd_mode       netstat        sleep
bash           echo           kill           nice           stat
bashbug        ed             linux32        pidof          stty
bbconfig       egrep          linux64        ping           su
busybox        false          ln             ping6          sync
cat            fatattr        login          pipe_progress  tar
catv           fdflush        ls             printenv       touch
chgrp          fgrep          lzop           ps             true
chmod          fsync          makemime       pwd            umount
chown          getopt         mkdir          reformime      uname
conspy         grep           mknod          rev            usleep
cp             gunzip         mktemp         rm             watch
date           gzip           more           rmdir          zcat
dd             hostname       mount          run-parts
df             ionice         mountpoint     sed
dmesg          iostat         mpstat         setserial
```
This image contains `sh` shell. Isn't it more standard?

Comment (aojea) [reply test/e2e/network/networking_perf.go:75]: nice catch
I've used `ash` because didn't know that alpine has `sh` :sweat_smile:

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Comment by aojea** [comment]
/assign @danwinship 
do you mind take a look for approval? 
it's a simple fix, the agnhost image has a binary as entrypoint, so we need to override it with a shell to be able to execute iperf inside.
I've tested the change locally

**Comment by cmluciano** [comment]
@aojea I don't have approver power but will try this out locally and give another review. Can you confirm how you ran this test locally? I'm gonna try a kubetest build against a cloud cluster.

**Comment by aojea** [comment]
> @aojea I don't have approver power but will try this out locally and give another review. Can you confirm how you ran this test locally? I'm gonna try a kubetest build against a cloud cluster.

I test it using kind https://github.com/kubernetes-sigs/kind/issues/1181#issuecomment-567001724

**Comment by danwinship** [comment]
> I've tested it locally, don't know if this is running in some CI

It probably runs in one of the periodic jobs, not something you can run from a PR. One way to get around that is to just push a temporary commit changing the tags on the test you want to run so it's no longer skipped in the default runs. But anyway, the fix looks right and you tested it.

/lgtm
/approve

**Pull request merged**